mitmproxy/proxy/protocol/websocket.py

```diff
@@ -81,27 +81,80 @@ def _handle_event(self, event, source_conn, other_conn, is_server):
         elif isinstance(event, events.PongReceived):
             return self._handle_pong_received(event, source_conn, other_conn, is_server)
         elif isinstance(event, events.ConnectionFailed):
-            return self._handle_connection_closed(event, source_conn, other_conn, is_server)
-        elif isinstance(event, events.ConnectionFailed):
             return self._handle_connection_failed(event)
+        elif isinstance(event, events.ConnectionClosed):
+            return self._handle_connection_closed(event, source_conn, other_conn, is_server)
 
         # fail-safe for unhandled events
         return True
 
     def _handle_data_received(self, event, source_conn, other_conn, is_server):
+        fb = self.server_frame_buffer if is_server else self.client_frame_buffer
+        fb.append(event.data)
+
+        if event.message_finished:
+            original_chunk_sizes = [len(f) for f in fb]
+            message_type = 0x01 if isinstance(event, events.TextReceived) else 0x02
```

Comment:
Please use https://github.com/python-hyper/wsproto/blob/24cb9bc5967e106684e19dd0afd1210aedbfc510/wsproto/frame_protocol.py#L58
At some point we might move our websocket stack into pathod, when mitmproxy no longer uses it.